public/pages/EditFeatures/containers/SampleAnomalies.tsx

```diff
@@ -74,6 +76,10 @@ export function SampleAnomalies(props: SampleAnomaliesProps) {
     endDate: initialEndDate.valueOf(),
   });
 
+  const [selectedHeatmapCell, setSelectedHeatmapCell] = useState<HeatmapCell>();
+
+  const isHCDetector = !isEmpty(get(newDetector, 'categoryField', []));
```

Comment:
minor: is adding the `categoryFields` prop necessary if it can already be retrieved via `detector` prop?

Reply:
if we get it from `detector` prop, it will always be false in case of single entity detector. 

If we retrieve it from `SampleAnomaliesProps`, when user already views Preview Result chart without enabling category field, the Preview Result Chart below will be automatically changed from line chart to heatmap when user enables/selects category field. 

In current implementation, in case of user already views Preview Result chart as single detector, only after user enables/selects category field, and clicks "Refresh", then result chart is changed to heatmap. This provides better UX.

Reply:
I see, this makes sense in the context of the sample anomalies page as part of the edit features page.